docker/entrypoint.sh

```diff
@@ -1,128 +1,292 @@
-            # Starting with a cached Zebra and lightwalletd tip, test all gRPC calls to lightwalletd, which calls Zebra.
-            ls -lh "$ZEBRA_CACHED_STATE_DIR"/*/* || (echo "No $ZEBRA_CACHED_STATE_DIR/*/*"; ls -lhR  "$ZEBRA_CACHED_STATE_DIR" | head -50 || echo "No $ZEBRA_CACHED_STATE_DIR directory")
-            ls -lhR "$LIGHTWALLETD_DATA_DIR/db" || (echo "No $LIGHTWALLETD_DATA_DIR/db"; ls -lhR "$LIGHTWALLETD_DATA_DIR" | head -50 || echo "No $LIGHTWALLETD_DATA_DIR directory")
-            cargo test --locked --release --features "$ENTRYPOINT_FEATURES" --package zebrad --test acceptance -- --nocapture --include-ignored lightwalletd_wallet_grpc_tests
-        elif [[ "$TEST_LWD_TRANSACTIONS" -eq "1" ]]; then
-            # Starting with a cached Zebra and lightwalletd tip, test sending transactions gRPC call to lightwalletd, which calls Zebra.
-            ls -lh "$ZEBRA_CACHED_STATE_DIR"/*/* || (echo "No $ZEBRA_CACHED_STATE_DIR/*/*"; ls -lhR  "$ZEBRA_CACHED_STATE_DIR" | head -50 || echo "No $ZEBRA_CACHED_STATE_DIR directory")
-            ls -lhR "$LIGHTWALLETD_DATA_DIR/db" || (echo "No $LIGHTWALLETD_DATA_DIR/db"; ls -lhR "$LIGHTWALLETD_DATA_DIR" | head -50 || echo "No $LIGHTWALLETD_DATA_DIR directory")
-            cargo test --locked --release --features "$ENTRYPOINT_FEATURES" --package zebrad --test acceptance -- --nocapture --include-ignored sending_transactions_using_lightwalletd
-
-        # These tests use mining code, but don't use gRPC.
-        # We add the mining feature here because our other code needs to pass tests without it.
-        elif [[ "$TEST_GET_BLOCK_TEMPLATE" -eq "1" ]]; then
-            # Starting with a cached Zebra tip, test getting a block template from Zebra's RPC server.
-            ls -lh "$ZEBRA_CACHED_STATE_DIR"/*/* || (echo "No $ZEBRA_CACHED_STATE_DIR/*/*"; ls -lhR  "$ZEBRA_CACHED_STATE_DIR" | head -50 || echo "No $ZEBRA_CACHED_STATE_DIR directory")
-            cargo test --locked --release --features "getblocktemplate-rpcs,$ENTRYPOINT_FEATURES" --package zebrad --test acceptance -- --nocapture --include-ignored get_block_template
-        elif [[ "$TEST_SUBMIT_BLOCK" -eq "1" ]]; then
-            # Starting with a cached Zebra tip, test sending a block to Zebra's RPC port.
-            ls -lh "$ZEBRA_CACHED_STATE_DIR"/*/* || (echo "No $ZEBRA_CACHED_STATE_DIR/*/*"; ls -lhR  "$ZEBRA_CACHED_STATE_DIR" | head -50 || echo "No $ZEBRA_CACHED_STATE_DIR directory")
-            cargo test --locked --release --features "getblocktemplate-rpcs,$ENTRYPOINT_FEATURES" --package zebrad --test acceptance -- --nocapture --include-ignored submit_block
-
-        else
-            exec "$@"
-        fi
+  --* | -*)
+    if [[ -n "${ZEBRA_CONF_PATH}" ]]; then
+        exec zebrad -c "${ZEBRA_CONF_PATH}" "$@" || { echo "Execution with custom configuration failed"; exit 1; }
+    else
+        exec zebrad "$@" || { echo "Execution failed"; exit 1; }
+    fi
+    ;;
+  "")
+    if [[ -n "${ZEBRA_CONF_PATH}" ]]; then
```

Comment:
It looks like the ZEBRA_CONF_PATH is always set, so the test workflow is running zebrad with the default config, rather than running the test commands below.

Maybe the test environment variable checks could go first, before we even build the default config?
(Generating the default config and logging it during the tests will make them harder to debug, because developers will assume the default config is being used by the tests.)

Reply:
Thanks! This might solve that: [ref: handle tests better if $ENTRYPOINT_FEATURES is set](https://github.com/ZcashFoundation/zebra/pull/7660/commits/93c4077792f3f68947cb26780d087f6def1bf05d)